### ISSUE TITLE
Cannot `docker run` in core-dev

### ISSUE THREAD

**Issue by kylos101** [gitpod-io/gitpod#7780]
### Bug description

Cannot run `docker run hello-world` in core-dev

### Steps to reproduce

1. Browse [here](https://main.staging.gitpod-dev.com/workspaces)
2. Try `docker run hello-world`
```shell
gitpod /workspace/template-python-flask $ docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
2db29710123e: Pull complete 
Digest: sha256:975f4b14f326b05db86e16de00144f9c12257553bba9484fed41f9b6f2257800
Status: Downloaded newer image for hello-world:latest
docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "proc" to rootfs at "/proc" caused: mount through procfd: operation not permitted: unknown.
ERRO[0003] error waiting for container: context canceled 
```

### Workspace affected

https://gitpodio-templatepythonf-5yc2qrpxba5.ws-dev.main.staging.gitpod-dev.com/

### Expected behavior

```shell
gitpod /workspace/gitpod (main) $ docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
2db29710123e: Pull complete 
Digest: sha256:975f4b14f326b05db86e16de00144f9c12257553bba9484fed41f9b6f2257800
Status: Downloaded newer image for hello-world:latest

Hello from Docker!
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
    (amd64)
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker ID:
 https://hub.docker.com/

For more examples and ideas, visit:
 https://docs.docker.com/get-started/
```

### Example repository

n/a

### Anything else?

This means Gitpodders will be unable to use docker in preview environments when testing.

**Comment by axonasif** [comment]
Hey @kylos101, I just tried and https://github.com/gitpod-io/gitpod/issues/7741 seems to be related.

**Comment by kylos101** [comment]
Hi @axonasif I believe #7741 is different, and that the [code change I referenced](https://github.com/gitpod-io/gitpod/issues/7741#issuecomment-1019524879) may help solve that issue. 

If I were to summarize this issue, the problem is our dev, which we fondly call `core-dev`, [lacks parity with production](https://12factor.net/dev-prod-parity). Phrased another way, we cannot effectively test our `docker in workspace` feature until this issue is solved.

One way to solve this issue is to build and test preview environments using the `with-vm` annotation/flag for runs in werft. The @gitpod-io/platform team mentioned that as a capability a couple weeks ago during an all-hands.

**Comment by axonasif** [comment]
Ooh, thanks for the insights and clarifying @kylos101 🙏

**Comment by axonasif** [comment]
> This is a test comment, will be deleted. Doing a test on frontapp 😅

**Comment by Furisto** [comment]
This is what the ws-daemon logs, when the error happens:
```
{"args":["/app/nsinsider","move-mount","--target","/tmp/supervisor1357977037/tmp/buildkit/runc-stargz/executor/xpqyzxmikgsdu6ze7ek6mfd6f/rootfs/proc","--pipe-fd","3"],"instanceId":"b399271c-53b2-4b74-87ec-a724ca85fcc6","level":"fatal","message":"no such file or directory","serviceContext":{"service":"nsinsider","version":""},"severity":"CRITICAL","time":"2022-01-26T12:14:42Z"}
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","error":"cannot move mount:\n    github.com/gitpod-io/gitpod/ws-daemon/pkg/iws.moveMount\n        github.com/gitpod-io/gitpod/ws-daemon/pkg/iws/iws.go:601\n  - cannot run nsinsider:\n    github.com/gitpod-io/gitpod/ws-daemon/pkg/iws.nsinsider\n        github.com/gitpod-io/gitpod/ws-daemon/pkg/iws/iws.go:718\n  - exit status 1","instanceId":"b399271c-53b2-4b74-87ec-a724ca85fcc6","level":"error","message":"cannot mount proc","procPID":924976,"reqPID":348,"serviceContext":{"service":"ws-daemon","version":"commit-b84ada30a98db180d94505406bf40a7faaba6756"},"severity":"ERROR","time":"2022-01-26T12:14:42Z","userId":"image-builder","workspaceId":"b399271c-53b2-4b74-87ec-a724ca85fcc6"}
```

**Comment by utam0k** [comment]
maybe related to #7793. I'll check.

**Comment by utam0k** [comment]
I made sure this issue related to the libseccomp issue. #7793
When I applied the libseccomp version upgrade[^1], docker worked fine. It will not work with `core-dev` because the linux kernel version must not be above 5.6.  So resolving issue #7793 will automatically fix this issue as well.

[^1]: https://github.com/gitpod-io/gitpod/commits/to/check-libseccomp

**Comment by kylos101** [comment]
This issue is resolved because of 5f9f701835292bbe28e71c8eb70f8a09965cb5ab

Thanks for the heads up, @Furisto !
